src/test/java/teammates/test/cases/BaseTestCaseWithDatastoreAccess.java

```diff
@@ -187,6 +187,8 @@ private void equalizeIrrelevantData(AccountAttributes expected, AccountAttribute
                 expected.studentProfile.googleId = actual.googleId;
             }
             expected.studentProfile.modifiedDate = actual.studentProfile.modifiedDate;
+            // data is expected to be sanitized
+            expected.sanitizeForSaving();
```

Comment:
Hmm could you explain more what you mean by self-sanitize? 
I believe you are concerned that making changes to the data in `expected` would be unexpected to people running this method? The `expected` in `equalizeIrrelevantData` for `AccountAttributes` and `InstructorAttributes` are actually copies of the given expected data. How about adding `toCopy` method for `CourseAttributes` as well?